### PR TITLE
[FileSystem] Added gsc functions

### PR DESCRIPTION
**Pull request by Joelrau**
gsc:

```
fileWrite(<filepath>, <string>, <mode>) example: fileWrite("NewFolder/example", "Example Text", "write"/"append")

fileRead(<filepath>) example: fileRead("NewFolder/example")

fileExists(<filepath>) example: fileExists("NewFolder/example")
```

### CODE REVIEW

File: src/Components/Modules/FileSystem.cpp [hunk 1]
```diff
@@ -340,6 +340,106 @@ namespace Components
...
 		Utils::Hook(0x642F60, FileSystem::IwdFreeStub, HOOK_CALL).install()->quick();
+
+		// Functions
+		Script::AddFunction("fileWrite", [](Game::scr_entref_t) // gsc: fileWrite(<filepath>, <string>, <mode>) example: fileWrite("NewFolder/example", "Example Text", "write"/"append")
+		{
+			auto path = Game::Scr_GetString(0);
+			auto text = Game::Scr_GetString(1);
+			auto mode = Game::Scr_GetString(2);
+
+			if (path == nullptr)
```
Comment (RektInator) [inline src/Components/Modules/FileSystem.cpp]: also add empty string check

File: src/Components/Modules/FileSystem.cpp [hunk 2]
```diff
@@ -340,6 +340,106 @@ namespace Components
...
+			{
+				Game::Com_Printf(0, "^1fileWrite: filepath not defined!\n");
+				return;
+			}
+			auto newpath = std::string(path) + ".iw4xFile";
+			std::vector<const char*> query_strings = { R"(..)", R"(../)", R"(..\)" };
+			for (size_t i = 0u; i < query_strings.size(); i++)
```
Comment (RektInator) [inline src/Components/Modules/FileSystem.cpp]: auto i = 0u

File: src/Components/Modules/FileSystem.cpp [hunk 3]
```diff
@@ -340,6 +340,106 @@ namespace Components
...
+			auto path = Game::Scr_GetString(0);
+			auto text = Game::Scr_GetString(1);
+			auto mode = Game::Scr_GetString(2);
+
+			if (path == nullptr)
+			{
+				Game::Com_Printf(0, "^1fileWrite: filepath not defined!\n");
+				return;
+			}
+			auto newpath = std::string(path) + ".iw4xFile";
```
Comment (RektInator) [inline src/Components/Modules/FileSystem.cpp]: auto newpath = path + ".iw4xFile"s;

File: src/Components/Modules/FileSystem.cpp [hunk 4]
```diff
@@ -340,6 +340,106 @@ namespace Components
...
+			std::vector<const char*> query_strings = { R"(..)", R"(../)", R"(..\)" };
```
Comment (RektInator) [inline src/Components/Modules/FileSystem.cpp]: keep naming consistent, I suggest queryStrings

File: src/Components/Modules/FileSystem.cpp [hunk 5]
```diff
@@ -340,6 +340,106 @@ namespace Components
...
+				}
+			}
+
+			if (mode == nullptr || mode != "append"s && mode != "write"s)
+			{
+				Game::Com_Printf(0, "^3fileWrite: mode not defined or was wrong, defaulting to 'write'\n");
+				mode = "write";
+			}
+
+			if (mode != nullptr)
```
Comment (RektInator) [inline src/Components/Modules/FileSystem.cpp]: useless if check

File: src/Components/Modules/FileSystem.cpp [hunk 6]
```diff
@@ -340,6 +340,106 @@ namespace Components
...
+			{
+				if (mode == "write"s)
+				{
+					auto fileHandle = Game::FS_FOpenFileWrite(newpath.data());
+					Game::FS_Write(text, strlen(text), fileHandle);
```
Comment (RektInator) [inline src/Components/Modules/FileSystem.cpp]: unneeded code duplication

**Comment by RaidMax** [comment]
Just a thought, but wouldn't it make sense to reimplement the original script file operations:
image

**Comment by Joelrau** [comment]
> Just a thought, but wouldn't it make sense to reimplement the original script file operations:
> image

i'd imagine it would make more sense, but as I am at the army right now I can't be the one to do it.

**Pull request closed without merging**